src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs

```diff
@@ -185,7 +185,74 @@ public override void WriteValue(object value)
         {
             ArgumentNullException.ThrowIfNull(value);
 
-            WriteString(XmlUntypedConverter.Untyped.ToString(value, _resolver));
+            Type sourceType = value.GetType();
+            if (!TypeScope.IsKnownType(sourceType))
+            {
+                WriteString(XmlUntypedConverter.Untyped.ToString(value, _resolver));
+                return;
+            }
+
+#pragma warning disable IL2026
+            TypeDesc typeDesc = s_typeScope.GetTypeDesc(sourceType);
+#pragma warning restore IL2026
+            switch (typeDesc.FormatterName)
+            {
+                case "Boolean":
+                    WriteWithBuffer((bool)value, XmlConvert.TryFormat);
+                    break;
+                case "Int32":
+                    WriteWithBuffer((int)value, XmlConvert.TryFormat);
+                    break;
+                case "Int16":
+                    WriteWithBuffer((short)value, XmlConvert.TryFormat);
+                    break;
+                case "Int64":
+                    WriteWithBuffer((long)value, XmlConvert.TryFormat);
+                    break;
+                case "Single":
+                    WriteWithBuffer((float)value, XmlConvert.TryFormat);
+                    break;
+                case "Double":
+                    WriteWithBuffer((double)value, XmlConvert.TryFormat);
+                    break;
+                case "Decimal":
+                    WriteWithBuffer((decimal)value, XmlConvert.TryFormat);
+                    break;
+                case "Byte":
+                    WriteWithBuffer((byte)value, XmlConvert.TryFormat);
+                    break;
+                case "SByte":
+                    WriteWithBuffer((sbyte)value, XmlConvert.TryFormat);
+                    break;
+                case "UInt16":
+                    WriteWithBuffer((ushort)value, XmlConvert.TryFormat);
+                    break;
+                case "UInt32":
+                    WriteWithBuffer((uint)value, XmlConvert.TryFormat);
+                    break;
+                case "UInt64":
+                    WriteWithBuffer((ulong)value, XmlConvert.TryFormat);
+                    break;
+                //Guid is not supported in XmlUntypedConverter.Untyped and throws exception
```

Comment:
This limitation is really strange, but we need to adapt XmlValueConverter to allow Guid in it's methods.